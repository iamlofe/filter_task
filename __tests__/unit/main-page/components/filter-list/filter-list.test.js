import React from 'react';
import { shallow } from 'enzyme';

import FilterList from 'main-page/components/filter-list/filter-list';

describe('account component', () => {
    it('is shallow rendered without crashing', () => {
        const baseProps = {
            isStatusLoadData: false,
            onLoadData: jest.fn(),
            onCreateDisplay: jest.fn(),
            filterIds: ['']
        };

        const filterChecbox = <FilterList {...baseProps} />;

        const filterChecboxComponent = shallow(filterChecbox);
        expect(filterChecboxComponent).toMatchSnapshot();
    });

    it('is shallow rendered with statusLoad true without crashing', () => {
        const baseProps = {
            onLoadData: jest.fn(),
            onCreateDisplay: jest.fn(),
            isStatusLoadData: false,
            filterIds: ['']
        };

        const filterChecbox = <FilterList {...baseProps} isStatusLoadData />;

        const filterChecboxComponent = shallow(filterChecbox);
        expect(filterChecboxComponent).toMatchSnapshot();
    });
});
