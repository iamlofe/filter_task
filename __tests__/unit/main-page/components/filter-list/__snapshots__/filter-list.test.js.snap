// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account component is shallow rendered with statusLoad true without crashing 1`] = `
<div
  className="filter-container__main-container"
>
  <span>
    ...
  </span>
  <div
    className="filter-container__state-container"
  >
    <withStateHandlers(Connect(withHandlers(pure(DisplayWidget))))
      filterId=""
      key=""
    />
  </div>
</div>
`;

exports[`account component is shallow rendered without crashing 1`] = `
<div
  className="filter-container__main-container"
>
  <FilterButtonOpenWidget
    className="filter__button-create"
    label="Create"
    onClick={[MockFunction]}
  />
  <div
    className="filter-container__state-container"
  >
    <withStateHandlers(Connect(withHandlers(pure(DisplayWidget))))
      filterId=""
      key=""
    />
  </div>
</div>
`;
